Microsoft.Azure.Cosmos.Encryption/src/AapServices/AapEncryptionProcessor.cs

```
@@ -0,0 +1,295 @@
+            using (JsonDocument document = JsonDocument.Parse(input))
+            {
+                JsonElement root = document.RootElement;
+                if (root.ValueKind != JsonValueKind.Object)
+                {
+                    throw new ArgumentException("Invalid document to encrypt", nameof(input));
+                }
+                else
+                {
+                    writer.WriteStartObject();
+                }
+
+                foreach (JsonProperty property in root.EnumerateObject())
+                {
+                    // nulls are not encrypted
+                    if (property.Value.ValueKind != JsonValueKind.Null &
+                        encryptionProperties != null)
+                    {
+                        if (encryptionProperties.EncryptedPaths != null &&
+                            encryptionProperties.EncryptedPaths.Any() &&
+                            encryptionProperties.EncryptedPaths.Contains('/' + property.Name))
+                        {
+                            try
+                            {
+                                (TypeMarker typeMarker, byte[] plainText) = Serialize(property.Value);
+
+                                byte[] cipherText = await encryptor.EncryptAsync(
+                                    plainText,
+                                    encryptionOptions.DataEncryptionKeyId,
+                                    encryptionOptions.EncryptionAlgorithm);
+
+                                byte[] cipherTextWithTypeMarker = new byte[cipherText.Length + 1];
+                                cipherTextWithTypeMarker[0] = (byte)typeMarker;
+                                Buffer.BlockCopy(cipherText, 0, cipherTextWithTypeMarker, 1, cipherText.Length);
+                                writer.WriteBase64String(property.Name, cipherTextWithTypeMarker);
+                            }
+                            catch (Exception ex)
```

Should we throw the exception instead of continuing?